src/url-loader/package-url-resolver.ts

```diff
@@ -39,7 +39,14 @@ export class PackageUrlResolver implements UrlResolver {
 
   canResolve(url: string): boolean {
     const urlObject = parseUrl(url);
-    const pathname = pathlib.normalize(decodeURI(urlObject.pathname || ''));
+    let decodedUrl;
+    try {
+      decodedUrl = decodeURI(urlObject.pathname || '');
+    } catch (e) {
+      return false;
```

Comment:
this is the kind of thing that makes me like checked exceptions, against all current attitudes...